sgf-viewer/src/components/subcomponents/GameList.svelte

```diff
@@ -14,13 +13,49 @@
         num_moves: "Game length",
         sgf_path: "SGF Path", // Should also include line number
     };
-    function indexToSgfPath(index: number) {
-        let keyIndex = Object.keys(tableColumns).indexOf("sgf_path");
-        let fileName = games[index][keyIndex].split("/").slice(-1)[0]; // Get last element
+    const sgfPathKeyIndex = Object.keys(tableColumns).indexOf("sgf_path");
+    $: sgfPaths = games.map((game) => {
+        const fileName = game[sgfPathKeyIndex].split("/").slice(-1)[0]; // Get last element
         return `/sgfs/${dirName}/${fileName}`;
+    })
+
+    const wgoPlayer = getContext("wgoPlayer");
+    const updateGame = getContext("updateGame");
+    let selectedRow: number = 0;
+    // Current game loaded by the wgoPlayer. Doesn't necessarily match with
+    // selectedRow if wgoPlayer is not yet initialized.
+    let currentSgfPath: string = "";
+    // Arbitrary large number, should be be larger than the move count in any of
+    // our SGFs. This is used to load the last move of the game.
+    const LAST_MOVE_NUM = 10000;
+    // Updates wgoPlayer and selectedRow.
+    $updateGame = (row, move = LAST_MOVE_NUM) => {
+        selectedRow = row;
+        if (!$wgoPlayer || row >= sgfPaths.length) {
+            return;
+        }
+        const newSgfPath = sgfPaths[row];
+        if (newSgfPath === currentSgfPath) {
+            $wgoPlayer.goTo(move);
+        } else {
+            $wgoPlayer.loadSgfFromFile(newSgfPath, move);
+            currentSgfPath = newSgfPath;
+        }
     }
-    $: if (games.length < selectedRow + 1) selectedRow = 0;
-    $: if (games.length > 0) sgfPath = indexToSgfPath(selectedRow);
+
+    let goPlayerIsInitialized: boolean = false;
+    $: if (!goPlayerIsInitialized && games.length > 0 && $wgoPlayer) {
+        let row = 0;
+        if (window.location.hash.startsWith("#" + dirName)) {
+            let searchParams = new URLSearchParams(window.location.search);
+            if (searchParams.has("row")) {
+                row = parseInt(searchParams.get("row"));
+            }
+        }
+        $updateGame(row);
+        goPlayerIsInitialized = true;
+    }
```

Comment:
now need some additional annoying logic to handle initialization, i.e., to call `updateGame` exactly once after `wgoPlayer` is created